src/server/scripts/Spells/spell_hunter.cpp

```diff
@@ -296,6 +297,69 @@ class spell_hun_disengage : public SpellScriptLoader
         }
 };
 
+// 56841 - Glyph of Arcane Shot
+class spell_hun_glyph_of_arcane_shot : public SpellScriptLoader
+{
+    public:
+        spell_hun_glyph_of_arcane_shot() : SpellScriptLoader("spell_hun_glyph_of_arcane_shot") { }
+
+        class spell_hun_glyph_of_arcane_shot_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_hun_glyph_of_arcane_shot_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_HUNTER_GLYPH_OF_ARCANE_SHOT))
+                    return false;
+                return true;
+            }
+
+            bool CheckProc(ProcEventInfo& eventInfo)
+            {
+                bool found = false;
+                Unit::AuraApplicationMap& Auras = eventInfo.GetProcTarget()->GetAppliedAuras();
+                for (Unit::AuraApplicationMap::const_iterator i = Auras.begin(); i != Auras.end(); ++i)
+                {
+                    Aura const* aura = i->second->GetBase();
+                    // Search only Serpent Sting, Viper Sting, Scorpid Sting, Wyvern Sting
+                    flag96 familyFlag = aura->GetSpellInfo()->SpellFamilyFlags;
+                    if (familyFlag.HasFlag(0xC000, 0x1080) && aura->GetSpellInfo()->SpellFamilyName == SPELLFAMILY_HUNTER)
+                    {
+                        found = true;
```

Comment:
You can remove the bool and change to a return true;
And replace return found by return false. Imo it looks cleaner